src/app/components/Arbitrage/Arbitrage.tsx

```diff
@@ -31,7 +31,16 @@ export function Arbitrage() {
   const opportunityArray = useMemo(
     () =>
       Object.values(data)
-        .filter(item => item.hasOwnProperty('rateToBalance'))
+        .filter(item => {
+          const rateToBalance = item?.rateToBalance;
```

Comment:
You can extract this whole check to a local function in this file (e.g. `isValidArbitrage`).